.circleci/config.yml

```
@@ -14,37 +14,37 @@ aliases:
 
   - &restore-node-modules-cache
     name: Restore node_modules cache
-    key: v1-node-modules-{{ checksum "yarn.lock" }}
+    key: v1-node-modules-{{ checksum "pnpm-lock.yaml" }}
 
-  - &restore-yarn-cache
-    name: Restore Yarn global cache
-    key: v1-yarn-global
+  - &restore-pnpm-cache
+    name: Restore pnpm store cache
+    key: v1-pnpm-store
 
   - &save-node-modules-cache
     name: Save node_modules cache
     paths:
       - node_modules
-    key: v1-node-modules-{{ checksum "yarn.lock" }}
+    key: v1-node-modules-{{ checksum "pnpm-lock.yaml" }}
 
-  - &save-yarn-cache
-    name: Save Yarn global cache
+  - &save-pnpm-cache
+    name: Save pnpm store cache
     paths:
-      # see https://yarnpkg.com/configuration/yarnrc#globalFolder
-      - /.yarn/berry/global
-    key: v1-yarn-global
+      - ~/.local/share/pnpm/store
+    key: v1-pnpm-store
 
 jobs:
   checkout-code:
     docker: *docker-node-image
     steps:
       - checkout
-      - restore_cache: *restore-yarn-cache
+      - restore_cache: *restore-pnpm-cache
       - restore_cache: *restore-node-modules-cache
       - run: sudo corepack enable
-      - run: yarn install --immutable
-      - run: yarn constraints
+      - run: pnpm install --frozen-lockfile
+      # TODO: Re-enable constraints check with pnpm equivalent if needed
+      # - run: pnpm constraints
       - save_cache: *save-node-modules-cache
-      - save_cache: *save-yarn-cache
+      - save_cache: *save-pnpm-cache
```

As dependencies change, old dependencies can be carried over from restored caches of previous CI builds. To prevent the pnpm store from snowballing in size, I usually recommend running `pnpm store prune` right before updating the store on CI.


```suggestion
      - run: pnpm store prune
      - save_cache: *save-pnpm-cache
```

---

Added as part of 9a8f406f3c8e48a2a389bdeb43af0e7f9796c386